packages/e2e/test/artillery/StakePoolSearch.yml

```diff
@@ -0,0 +1,20 @@
+config:
+  target: 'http://localhost:4000'
```

Comment:
Thank you @IvayloAndonov .

About the `target` attribute:
- it is actually a refuse, after the today refactoring it is no longer used (need to test if artillery works without it or it is a required field)
- I agree: if required, any solution is better than an hard coded value!

About the 2nd proposal, the initial check: I'll do that.

Reply:
Hi @IvayloAndonov , @rhyslbw and @iadmytro ,

441952f06 should solve this, but please let me give you some clarifications.

`target` is a required field in artillery configuration: I simply used `http://localhost` as a placeholder.

I thought the most appropriate method to perform the check you suggested is actually using the `healthCheck` method of the `provider`. I searched a lot but it seems artillery doesn't offer a _phase level hook_, it offers only _scenario level hooks_. Please look at [this request](https://github.com/artilleryio/artillery/issues/131).

In artillery, phases are executed only once and generates several virtual users; scenarios are executed once for each generated virtual user.

I had two options:
1. to write a script to perform the check externally from artillery and running this one just in case
2. perform the `healthCheck` at the beginning of the scenario

I chosen the second one for more than one reason: robustness, cheaper to be written, makes the test flow more compliant with our sdk use pattern...

Reply:
The second option sounds good to me as well. Just some extra explanation about the scenario definition in [StakePoolSearch.yml](https://github.com/input-output-hk/cardano-js-sdk/pull/371/files#diff-2a2837d6babf0035796a40acb10cad57a8145407840d9870c9799294fd055a27) would be good (how the nested 2 loops work) but this is not a requested change by my side. Thanks for the clarifications! @iccicci

Reply:
Hi @IvayloAndonov ,
defaf32ae should answer to your question.